src/datasets/arrow_dataset.py

```diff
@@ -5138,6 +5148,34 @@ def push_to_hub(
                 "Failed to push_to_hub: please specify either max_shard_size or num_shards, but not both."
             )
 
+        # Create repo if it doesn't exist
+        api = HfApi(endpoint=config.HF_ENDPOINT)
+        try:
+            api.repo_info(repo_id, token=token)
+        except RepositoryNotFoundError:
+            repo_url = create_repo(hf_api=api, repo_id=repo_id, token=token, exist_ok=True, repo_type="dataset")
+            repo_id = repo_url.repo_id
+
+        # Try to find PR branch if branch is supplied
+        if create_pr and branch is not None:
+            for discussion in get_repo_discussions(repo_id, repo_type="dataset"):
```

Comment:
Actually, this has been introduced in [huggingface 0.9 release](https://github.com/huggingface/huggingface_hub/releases/tag/v0.9.0). @albertvillanova  @lhoestq how do you deal with stuff like this ?

Reply:
For the `get_repo_discussions`-problem, maybe do like with `upload_file` in `utils._hf_hub_fixes.py`, i.e.:
```
def get_repo_discussions(
    hf_api: HfApi,
    repo_id: str,
    repo_type: Optional[str] = None,
    token: Optional[str] = None,
) -> Iterator[Discussion]:
    """
    The method `huggingface_hub.HfApi.get_repo_discussions` was introduced in 0.9.0.
    """
    if version.parse(huggingface_hub.__version__) < version.parse("0.9.0"):
        raise TypeError(
            "The method `get_repo_discussions` was introduced in huggingface_hub 0.9.0. Please update huggingface_hub to >=0.9.0 to use this method."
        )
    else:
        return hf_api.get_repo_discussions(
            repo_id=repo_id,
            repo_type=repo_type,
            token=token,
        )
```

Reply:
Hi ! I think you can check for `huggingface_hub` version directly in `push_to_hub` and provide a more specialized error message:
```python
"Using `create_pr` requires `huggingface-hub>=0.9.0`. Please use a more recent version of `huggingface-hub`."
```

Reply:
Like [this](https://github.com/huggingface/datasets/pull/5528/commits/c7a998f28135c1a68d6e603fb036c5776225c6ab) ?

Reply:
@lhoestq I remember we decided to put all huggingface_hub version-dependence logic in one single place (file `_hf_hub_fixes.py`). The reason was to make things easier once we decide to stop supporting older hfh versions...

Reply:
Yea I guess it'b be nice to still have the function in `_hf_hub_fixes`. But having the specialized error is necessary to not be confusing for users

Reply:
@lhoestq: https://github.com/huggingface/datasets/pull/4106#issuecomment-1090283193

> adrinjalali: I find it good practice to have all dependency version related code in a single file so that when you decide to remove support for an old version of a dependency it's easy to find and remove them

> lhoestq: good idea, thanks !

> Commit: [move version specific code to its own file](https://github.com/huggingface/datasets/pull/4106/commits/399aaf47113eed93d9b786e822064584505969c3)

Reply:
Ill move the function back into `_hf_hub_fixes` and leave the error message.